src/types.rs

```diff
@@ -23,11 +23,17 @@ impl<'a> From<&'a str> for CompleteStr<'a> {
   }
 }
 
+impl<'a, 'b> From<&'b &'a str> for CompleteStr<'a> {
+  fn from(src: &'b &'a str) -> Self {
+    CompleteStr(*src)
+  }
+}
+
 impl<'a> Deref for CompleteStr<'a> {
-  type Target = str;
+  type Target = &'a str;
 
-  fn deref(&self) -> &str {
-    self.0
+  fn deref(&self) -> &Self::Target {
+    &self.0
```

Comment:
The value returned from `Deref::deref` is now an `&'_ &'a str`, suitable for reconstruction into a `CompleteStr<'a>` via the `From` impl above. Formerly, the value returned was an `&'_ str` where the reference lifetime was tied to the *newtype wrapper*, not to the *referent str*, which was impossible to unify during `From`.